collector/bgp_test.go

```diff
@@ -428,35 +428,41 @@ func TestProcessBgpL2vpnEvpnSummary(t *testing.T) {
 }
 
 func TestProcessBGPPeerDesc(t *testing.T) {
-	expectedJSONOutput := make(map[string]map[string]map[string]string)
-	expectedJSONOutput["default"] = make(map[string]map[string]string)
-	expectedJSONOutput["default"]["10.1.1.10"] = make(map[string]string)
-	expectedJSONOutput["default"]["10.1.1.10"]["desc"] = "rt1"
-	expectedJSONOutput["default"]["swp2"] = make(map[string]string)
-	expectedJSONOutput["default"]["swp2"]["desc"] = "fw1"
-	expectedJSONOutput["vrf1"] = make(map[string]map[string]string)
-	expectedJSONOutput["vrf1"]["10.2.0.1"] = make(map[string]string)
-	expectedJSONOutput["vrf1"]["10.2.0.1"]["desc"] = "remote"
+	// expectedDescOutput := make(map[string]map[string]map[string]string)
+	// expectedJSONOutput["default"] = make(map[string]map[string]string)
+	// expectedJSONOutput["default"]["10.1.1.10"] = make(map[string]string)
+	// expectedJSONOutput["default"]["10.1.1.10"]["desc"] = "rt1"
+	// expectedJSONOutput["default"]["swp2"] = make(map[string]string)
+	// expectedJSONOutput["default"]["swp2"]["desc"] = "fw1"
+	// expectedJSONOutput["vrf1"] = make(map[string]map[string]string)
+	// expectedJSONOutput["vrf1"]["10.2.0.1"] = make(map[string]string)
+	// expectedJSONOutput["vrf1"]["10.2.0.1"]["desc"] = "remote"
 
-	expectedTextOutput := make(map[string]map[string]string)
-	expectedTextOutput["default"] = make(map[string]string)
-	expectedTextOutput["default"]["10.1.1.10"] = "{\"desc\":\"rt1\"}"
-	expectedTextOutput["default"]["swp2"] = "{\"desc\":\"fw1\"}"
-	expectedTextOutput["vrf1"] = make(map[string]string)
-	expectedTextOutput["vrf1"]["10.2.0.1"] = "{\"desc\":\"remote\"}"
+	bgpNeighborsDefault := make(map[string]bgpBGPNeighborsNeighbor)
+	bgpNeighborsDefault["10.1.1.10"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"rt1\"}"}
+	bgpNeighborsDefault["swp2"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"fw1\"}"}
 
-	descJSON, descText, err := processBGPPeerDesc(nil, bgpNeighborDesc)
-	if err != nil {
-		t.Errorf("error calling processBGPPeerDesc: %s", err)
+	bgpNeighborsVRF := make(map[string]bgpBGPNeighborsNeighbor)
+	bgpNeighborsVRF["10.2.0.1"] = bgpBGPNeighborsNeighbor{Desc: "{\"desc\":\"remote\"}"}
+
+	expectedOutput := make(map[string]bgpBGPNeighborsVRF)
+	expectedOutput["default"] = bgpBGPNeighborsVRF{
+		ID:           0,
+		Name:         "default",
+		BGPNeighbors: bgpNeighborsDefault,
+	}
+	expectedOutput["vrf1"] = bgpBGPNeighborsVRF{
+		ID:           -1,
+		Name:         "vrf1",
+		BGPNeighbors: bgpNeighborsVRF,
```

Comment:
This can be written a bit more concisely as a struct literal, e.g.
```go
	expectedOutput := map[string]bgpBGPNeighborsVRF{
		"default": {
			ID:   0,
			Name: "default",
			BGPNeighbors: map[string]bgpBGPNeighborsNeighbor{
				"10.1.1.10": {Desc: "{\"desc\":\"rt1\"}"},
				"swp2":      {Desc: "{\"desc\":\"fw1\"}"},
			},
		},
		"vrf1": {
			ID:   -1,
			Name: "vrf1",
			BGPNeighbors: map[string]bgpBGPNeighborsNeighbor{
				"10.2.0.1": {Desc: "{\"desc\":\"remote\"}"},
			},
		},
	}
```

Reply:
Fixed